### PR TITLE
fix: team deletion

### PR DESCRIPTION
**Pull request by macobo**
When deleting teams noticed we issue ALTER TABLES with team_id = NULL. This should fix this issue

## How did you test this code?

I didn't - quick PR so I don't forget this. :(

### CODE REVIEW

File: posthog/api/team.py [hunk 1]
```diff
@@ -190,8 +190,8 @@ def get_object(self):
         return team
 
     def perform_destroy(self, team: Team):
-        super().perform_destroy(team)
         delete_clickhouse_data.delay(team_ids=[team.pk])
+        super().perform_destroy(team)
```
Comment (hazzadous) [inline posthog/api/team.py:194]: non-blocking

I guess there is a case where perform_destroy fails but we delete from ClickHouse anyway. Might be better to 

```suggestion
        team_id = team.pk
        super().perform_destroy(team)
        delete_clickhouse_data.delay(team_ids=[team_id])
```

But probably not super important

**Review by hazzadous — APPROVED** [review]
Looks good

Given that this is now going to start deleting stuff, do we have to consider:

 1. load on ClickHouse
 2. if it's actually deleting the right stuff, i.e. is it tested elsewhere?

File: posthog/api/team.py (hunk 1 above)
Comment (macobo) [reply posthog/api/team.py:194]: Very good point to be honest. Let's go with that!

**Comment by macobo** [comment]
do we have to consider:

> load on ClickHouse

Yep. This is expensive. We should really be batching these deletes together and run them on a weekend, but since we delete the Teams (not mark them as is_deleted) this is slightly tricky to implement.

>if it's actually deleting the right stuff, i.e. is it tested elsewhere?

It is! Organization deletion works today and the task is itself tested.

**Pull request merged**